float64[2] targetVector
float64[2] wallVector
